lib/repl.js

```diff
@@ -585,10 +595,21 @@ REPLServer.prototype.createContext = function() {
     context = global;
   } else {
     context = vm.createContext();
-    for (var i in global) context[i] = global[i];
-    context.console = new Console(this.outputStream);
     context.global = context;
     context.global.global = context;
+    const _console = new Console(this.outputStream);
+    Object.defineProperty(context, 'console', {
+      configurable: true,
+      enumerable: true,
+      get: () => _console
+    });
+    Object.getOwnPropertyNames(global).filter((name) => {
+      if (name === 'console' || name === 'global') return false;
+      return GLOBAL_OBJECT_PROPERTY_MAP[name] === undefined;
```

Comment:
@lance sorry for not following up here…

Yes, `global` has no “own” `constructor` property unless set by user code, but `GLOBAL_OBJECT_PROPERTY_MAP['constructor']` *will* be truthy whereas `GLOBAL_OBJECT_PROPERTIES.indexOf('constructor') === -1` will not, so these are in fact not equivalent.

I’m not thinking about tab completion here, I’m just wondering if these properties should or should not be copied to the new context.